zebra-chain/src/sapling/commitment.rs

```diff
@@ -297,6 +293,80 @@ impl ValueCommitment {
     }
 }
 
+/// A Homomorphic Pedersen commitment to the value of a note, used in Spend and
+/// Output descriptions.
+///
+/// Elements that are of small order are not allowed. This is a separate
+/// consensus rule and not intrinsic of value commitments; which is why this
+/// type exists.
+///
+/// This is denoted by `cv` in the specification.
+///
+/// https://zips.z.cash/protocol/protocol.pdf#spenddesc
+/// https://zips.z.cash/protocol/protocol.pdf#outputdesc
+#[derive(Debug, Clone, Copy, Deserialize, PartialEq, Eq, Serialize)]
+pub struct NotSmallOrderValueCommitment(ValueCommitment);
```

Comment:
Not the prettiest ever but does the job and is Nice™ from a type semantics standpoint. 👍